gluonnlp/model/convolutional_encoder.py

```diff
@@ -131,7 +133,13 @@ def __init__(self,
                 self._projection = None
                 self._output_size = maxpool_output_size
 
-    def forward(self, inputs, mask=None): # pylint: disable=arguments-differ
+    def __call__(self, inputs, mask=None):
+        if mask is not None:
+            return super(ConvolutionalEncoder, self).__call__(inputs, mask)
+        else:
+            return super(ConvolutionalEncoder, self).__call__(inputs)
+
+    def hybrid_forward(self, F, inputs, mask=None): # pylint: disable=arguments-differ
```

Comment:
remove none default.

Reply:
then we can do away with the override of `__call__`

Reply:
I don't think so, because for HybridBlock `None` is not a valid input: `E       AssertionError: HybridBlock input must be (nested) list of Symbol or NDArray, but got None of type <class 'NoneType'>`

Reply:
Ie. to allow `encoder(inputs, None)` we must override __call__ or propose a change in mxnet if you believe this is actually not intended.

Reply:
Right, I met the exact same situation in gluon RNN layer. The trick is to use positional arguments for hybrid_forward such as `hybrid_forward(self, F, *inputs)`

Reply:
Even with positional arguments I get`E AssertionError: HybridBlock input must be (nested) list of Symbol or NDArray, but got None of type <class 'NoneType'>`. The reason I overwrote `__call__` is to preserve the current behavior where users can set mask explicitly to None, ie. call: `encoder(inputs, None)` compared to `encoder(inputs)` where the mask just defaults to None. If you are fine with only supporting the latter we can remove the `__call__` overwrite.
`encoder(inputs, mask)` with mask being a proper NDArray is always possible.

Reply:
We can drop the support for encoder(inputs, None).